src/helpers/lockdrop/BitcoinLockdrop.ts

```diff
@@ -222,27 +274,35 @@ export function btcLockScript(publicKeyHex: string, blockSequence: number): Buff
  * creates a P2SH instance that locks the sent token for the given duration.
  * the locked tokens can only be claimed by the provided public key
  * @param lockDays the lock duration in days
- * @param publicKey uncompressed public key of the locker
+ * @param publicKey public key of the locker. This can be both compressed or uncompressed
  * @param network bitcoin network the script will generate for
  */
-export function getLockP2SH(lockDays: number, publicKey: string, network: BtcNetwork) {
-    const netType = network === BtcNetwork.MainNet ? bitcoinjs.networks.bitcoin : bitcoinjs.networks.testnet;
-
+export function getLockP2SH(lockDays: number, publicKey: string, network: bitcoinjs.Network) {
     return bitcoinjs.payments.p2sh({
-        network: netType,
+        network: network,
         redeem: {
-            output: btcLockScript(publicKey, daysToBlocks(lockDays)),
+            output: btcLockScript(publicKey, daysToBlockSequence(lockDays), network),
         },
     });
 }
 
+/**
+ * creates a lock redeem UTXO
+ * @param signer the signer for signing the transaction hash
+ * @param network network type (bitcoinjs-lib)
+ * @param lockTx the transaction that locks the value to P2SH address
+ * @param lockScript the lock script (P2SH)
+ * @param blockSequence block sequence to lock the funds, should be the same value used in the lock script
+ * @param recipient recipient for the transaction output
+ * @param fee transaction fee for the lock transaction
+ */
 export function btcUnlockTx(
     signer: Signer,
     network: Network,
     lockTx: UnspentTx,
     lockScript: Buffer,
-    lockBlocks: number,
-    recipient: string,
+    blockSequence: number,
+    recipientAddress: string,
     fee: number, // satoshis
```

Comment:
In the function btcUnlockTx(), the parameters lockBlocks and fee should be ensured to be non-negative.

Reply:
fixed in https://github.com/staketechnologies/lockdrop-ui/pull/63/commits/2cbb55127419765528ee9d6c771812f5d63ca12d, please check